packages/perf-react/src/lexical/plugins/ContextMenuPlugin.tsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
+import { LexicalContextMenuPlugin, MenuOption } from "@lexical/react/LexicalContextMenuPlugin";
+import { type LexicalNode, COPY_COMMAND, CUT_COMMAND, PASTE_COMMAND } from "lexical";
+import { useCallback, useMemo } from "react";
+import * as ReactDOM from "react-dom";
+
+function ContextMenuItem({
+  index,
+  isSelected,
+  onClick,
+  onMouseEnter,
+  option,
+}: {
+  index: number;
+  isSelected: boolean;
+  onClick: () => void;
+  onMouseEnter: () => void;
+  option: ContextMenuOption;
+}) {
+  let className = "item";
+  if (isSelected) {
+    className += " selected";
+  }
+  return (
+    <li
+      key={option.key}
+      tabIndex={-1}
+      className={className}
+      ref={option.setRefElement}
+      role="option"
+      aria-selected={isSelected}
+      id={"typeahead-item-" + index}
+      onMouseEnter={onMouseEnter}
+      onClick={onClick}
+    >
+      <span className="text">{option.title}</span>
+    </li>
+  );
+}
+
+function ContextMenu({
+  options,
+  selectedItemIndex,
+  onOptionClick,
+  onOptionMouseEnter,
+}: {
+  selectedItemIndex: number | null;
+  onOptionClick: (option: ContextMenuOption, index: number) => void;
+  onOptionMouseEnter: (index: number) => void;
+  options: Array<ContextMenuOption>;
+}) {
+  return (
+    <div className="typeahead-popover">
+      <ul>
+        {options.map((option: ContextMenuOption, i: number) => (
+          <ContextMenuItem
+            index={i}
+            isSelected={selectedItemIndex === i}
+            onClick={() => onOptionClick(option, i)}
+            onMouseEnter={() => onOptionMouseEnter(i)}
+            key={option.key}
+            option={option}
+          />
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+export class ContextMenuOption extends MenuOption {
+  title: string;
+  onSelect: (targetNode: LexicalNode | null) => void;
+  constructor(
+    title: string,
+    options: {
+      onSelect: (targetNode: LexicalNode | null) => void;
+    },
+  ) {
+    super(title);
+    this.title = title;
+    this.onSelect = options.onSelect.bind(this);
+  }
+}
+
+export default function ContextMenuPlugin(): JSX.Element {
+  const [editor] = useLexicalComposerContext();
+
+  const options = useMemo(() => {
+    return [
+      new ContextMenuOption(`Copy`, {
+        onSelect: (_node) => {
+          editor.dispatchCommand(COPY_COMMAND, null);
+        },
+      }),
+      new ContextMenuOption(`Cut`, {
+        onSelect: (_node) => {
+          editor.dispatchCommand(CUT_COMMAND, null);
+        },
+      }),
+      new ContextMenuOption(`Paste`, {
```

Comment:
Curiously in [Codesandbox](https://codesandbox.io/p/github/BiblioNexus-Foundation/scripture-editors/context-menu) (which I think is Linux) the context menu _Paste_ item doesn't paste (_Ctrl + V_ works fine though). Running locally on my Win11 the context paste works just fine. Can you investigate this?

Reply:
This appears to be working now.